### PR TITLE
fix: rolling and cum operator on multiple series

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY

Currently, `Rolling Window` calculation results are incorrect on multiple series. 

Frontend codes at: https://github.com/apache-superset/superset-ui/pull/1386

image

#### After
image

#### Before
image

### TESTING INSTRUCTIONS

1. select `birth_names` dataset
2. switch to `time-series` chart
3. set time grain to `year`
4. set time range to `1970 - 2000`
5. set metrics to `sum__num`
6. set group by to `gender`
8. set rolling function to `sum`
9. set periods to `2`
10. set min periods to `2`
9. use the south panel to observe the accuracy of the data

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by villebro — COMMENTED** [review]
Could we add a quick unit test for `is_pivot_df = False` to ensure this doesn't break later?

**Comment by zhaoyongjie** [comment]
> Could we add a quick unit test for `is_pivot_df = False` to ensure this doesn't break later?

Sure, I will add such UT.

File: superset/utils/pandas_postprocessing.py [hunk 1]
```diff
@@ -405,10 +417,20 @@ def rolling(  # pylint: disable=too-many-arguments
...
             )
         ) from ex
-    df = _append_columns(df, df_rolling, columns)
+
+    if is_pivot_df:
+        agg_in_pivot_df = df.columns.get_level_values(0).drop_duplicates().to_list()
+        agg: Dict[str, Dict[str, Any]] = {col: {} for col in agg_in_pivot_df}
+        df_rolling.columns = [
+            _flatten_column_after_pivot(col, agg) for col in df_rolling.columns
+        ]
```
Comment (zhaoyongjie) [inline superset/utils/pandas_postprocessing.py:426]: I will create a separate PR for refactoring _flatten_column_after_pivot

**Review by villebro — APPROVED** [review]
LGTM, thanks for this fix+added code coverage! 🎉

**Pull request merged**